File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Stack {
    private final int[] stack;
    private int top;

    // 생성자
    public Stack(int N) {
        stack = new int[N];
        top = 0;
    }

    public void push(int x) {
        stack[top++] = x;
    }

    public int pop() {
        return top < 1 ? -1 : stack[--top];
    }

    public int size() {
        return top;
    }

    public int isEmpty() {
        return top == 0 ? 1 : 0;
    }

    public int top() {
        return top < 1 ? -1 : stack[top - 1];
    }
}

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int N = Integer.parseInt(br.readLine());
        Stack stack = new Stack(N);

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            switch (st.nextToken()) {
                case "push":
                    stack.push(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    System.out.println(stack.pop());
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "empty":
                    System.out.println(stack.isEmpty());
                    break;
                case "top":
                    System.out.println(stack.top());
                    break;
            }
        }
        br.close();
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.util.Stack

fun main() {

    // 첫째 줄에 주어지는 명령의 수 N (1 ≤ N ≤ 10,000)이 주어진다.
    val N = readln().toInt()
    val stack = Stack<Int>()

    // 둘째 줄부터 N개의 줄에는 명령이 하나씩 주어진다.

    repeat(N) {
        /*
        * push X: 정수 X를 스택에 넣는 연산이다.
        * pop: 스택에서 가장 위에 있는 정수를 빼고, 그 수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        * size: 스택에 들어있는 정수의 개수를 출력한다.
        * empty: 스택이 비어있으면 1, 아니면 0을 출력한다.
        * top: 스택의 가장 위에 있는 정수를 출력한다. 만약 스택에 들어있는 정수가 없는 경우에는 -1을 출력한다.
        * */
        val line = readln()
        val parts = line.split(' ') // 명령과 인자를 분리

        when (parts[0]) {
            "push" -> {
                val x = parts[1].toInt() // 두 번째 토큰을 정수로 변환
                stack.push(x)
            }
            "pop" -> {
                if (stack.isEmpty()) println(-1) else println(stack.pop())
            }
            "size" -> {
                println(stack.size)
            }
            "empty" -> {
                if (stack.isEmpty()) println(1) else println(0)
            }
            "top" -> {
                if (stack.isEmpty()) println(-1) else println(stack.last())
            }
        }

    }
}